flowmachine/flowmachine/features/subscriber/home_location_monthly.py

```diff
@@ -0,0 +1,198 @@
+from typing import List, Union
+from flowmachine.core.spatial_unit import AnySpatialUnit
+from datetime import datetime, timedelta
+from flowmachine.core.query import Query
+from dateutil.relativedelta import relativedelta
+from dateutil.rrule import rrule, DAILY
+from copy import copy
+
+from flowmachine.features.subscriber.daily_location import daily_location
+from flowmachine.features.subscriber.modal_location import ModalLocation
+from flowmachine.features.subscriber.unique_active_subscribers import (
+    UniqueActiveSubscribers,
+    ActiveSubscribers,
+)
+from flowmachine.features.subscriber.last_location import LastLocation
+from flowmachine.features.subscriber.location_visits import LocationVisits
+
+
+class HomeLocationMonthly(Query):
```

Comment:
I'm not keen on the name 'HomeLocation' - there are many different ways home location could be estimated, so it would be good to have a more descriptive name. I haven't come up with a better name yet, though - I'll come back to you on that one.

(Sorry, I realise the name came from my diagram...)

Reply from @Thingus:
> How about `ReferencedModalThresholdLocation`?